pylint/utils.py

```diff
@@ -433,63 +439,129 @@ def print_full_documentation(self):
                             title = 'General options'
                         else:
                             title = '%s options' % section.capitalize()
-                        print(title)
-                        print('~' * len(title))
-                        _rest_format_section(sys.stdout, None, options)
-                        print("")
+                        print(title, file=stream)
+                        print('~' * len(title), file=stream)
+                        _rest_format_section(stream, None, options)
+                        print("", file=stream)
             else:
+                name = checker.name
                 try:
-                    by_checker[checker.name][0] += checker.options_and_values()
-                    by_checker[checker.name][1].update(checker.msgs)
-                    by_checker[checker.name][2] += checker.reports
+                    by_checker[name]['options'] += checker.options_and_values()
+                    by_checker[name]['msgs'].update(checker.msgs)
+                    by_checker[name]['reports'] += checker.reports
                 except KeyError:
-                    by_checker[checker.name] = [list(checker.options_and_values()),
-                                                dict(checker.msgs),
-                                                list(checker.reports)]
-
-        print("Pylint checkers' options and switches")
-        print("-------------------------------------")
-        print("")
-        print("Pylint checkers can provide three set of features:")
-        print("")
-        print("* options that control their execution,")
-        print("* messages that they can raise,")
-        print("* reports that they can generate.")
-        print("")
-        print("Below is a list of all checkers and their features.")
-        print("")
+                    by_checker[name] = {
+                        'options': list(checker.options_and_values()),
+                        'msgs':    dict(checker.msgs),
+                        'reports': list(checker.reports),
+                    }
+
+        print("Pylint checkers' options and switches", file=stream)
+        print("-------------------------------------", file=stream)
+        print("", file=stream)
+        print("Pylint checkers can provide three set of features:", file=stream)
+        print("", file=stream)
+        print("* options that control their execution,", file=stream)
+        print("* messages that they can raise,", file=stream)
+        print("* reports that they can generate.", file=stream)
+        print("", file=stream)
+        print("Below is a list of all checkers and their features.", file=stream)
+        print("", file=stream)
+
+        for checker, info in six.iteritems(by_checker):
+            self._print_checker_doc(checker, info, stream=stream)
+
+    def _print_checker_doc(self,  # pylint: disable=no-self-use
```

Comment:
This can be usually alleviated through a ``@staticmethod`` decorator, so you don't need the disable here.